src/Fabulous.Core/ViewConverters.fs

```diff
@@ -452,6 +464,12 @@ module Converters =
                 oc
         updateCollectionGeneric (ValueOption.map seqToArray prevCollOpt) (ValueOption.map seqToArray collOpt) targetColl ListElementData (fun _ _ _ -> ()) canReuseChild (fun _ curr target -> target.Key <- curr) 
 
+    /// Update the items in a SearchHandler control, given previous and current view elements
+    let updateSearchHandlerItems (prevCollOpt: seq<'T> voption) (collOpt: seq<'T> voption) (target: Xamarin.Forms.SearchHandler) = 
```

Comment:
Had to replace the ItemsSource everytime, because removing items from ObservableCollection here has no effect...

Reply:
I will try to reproduce this behavior of ObservableCollection in C#. If the problem is really here, I'll open an issue on Xamarin.Forms

Reply:
Opened an issue for that: https://github.com/xamarin/Xamarin.Forms/issues/6500